File: test/samples/emptylines/emptylines.pas
program EmptyLines;

type
    { a char struct }
    StructWithAChar = record
        Char1: char; { a char }
        Int2: longint; { an integer }
    end;

    { a struct }
    A = record
        ListInStruct: array of longint; { a list in a struct }
        StructInStruct: StructWithAChar; { a struct in a struct }
    end;

    { a sized struct }
    SizedStruct = record
        Size: longint; { the size }
        StringInStruct: string; { the string }
    end;

    T_EmptyList = array of longint;
    T_EmptyInSample = array of longint;

{ @param EmptyList an empty list }
{ @param BufferString here to check correct parsing of empty line above }
{ @param N an integer, will be 0 in the sample input }
{ @param EmptyInSample an empty list (only in the sample) }
{ @param EmptyString an empty string }
{ @param Main an other buffer string }
{ @param EmptyCharList an empty char list }
{ @param NonEmptyCharList an char list, non empty }
{ @param StructWithEmptyLine a struct containing an empty line, then a struct }
{ @param ASizedStruct a sized struct containing an empty line }
{ @param Finish a string to finish }
procedure EmptyLines(const EmptyList: T_EmptyList; const BufferString: string; N: longint; const EmptyInSample: T_EmptyInSample; const EmptyString: string; const Main: string; const EmptyCharList: string; const NonEmptyCharList: string; const StructWithEmptyLine: A; const ASizedStruct: SizedStruct; const Finish: string);
begin
    {* TODO Wow, lots of empty lines! *}
end;

var
    EmptyList: T_EmptyList; { an empty list }
    BufferString: string; { here to check correct parsing of empty line above }
    N: longint; { an integer, will be 0 in the sample input }
    EmptyInSample: T_EmptyInSample; { an empty list (only in the sample) }
    EmptyString: string; { an empty string }
    Main: string; { an other buffer string }
    EmptyCharList: string; { an empty char list }
    NonEmptyCharList: string; { an char list, non empty }
    StructWithEmptyLine: A; { a struct containing an empty line, then a struct }
    ASizedStruct: SizedStruct; { a sized struct containing an empty line }
    Finish: string; { a string to finish }
    i: longint;
begin
    setLength(EmptyList, 0);
    for i := 0 to 0 - 1 do
        read(EmptyList[i]);
    readln();
    readln(BufferString);
    readln(N);
    setLength(EmptyInSample, N);
    for i := 0 to N - 1 do
        read(EmptyInSample[i]);
    readln();
    readln(EmptyString);
    readln(Main);
    readln(EmptyCharList);
    readln(NonEmptyCharList);
    setLength(StructWithEmptyLine.ListInStruct, N);
    for i := 0 to N - 1 do
        read(StructWithEmptyLine.ListInStruct[i]);
    readln();
    readln(StructWithEmptyLine.StructInStruct.Char1, StructWithEmptyLine.StructInStruct.Int2);
    readln(ASizedStruct.Size);
    readln(ASizedStruct.StringInStruct);
    readln(Finish);
    EmptyLines(EmptyList, BufferString, N, EmptyInSample, EmptyString, Main, EmptyCharList, NonEmptyCharList, StructWithEmptyLine, ASizedStruct, Finish);
end.

File: test/samples/emptylines/emptylines.go
package main

import "bufio"
import "fmt"
import "os"
import "strconv"
import "strings"

// a char struct
type StructWithAChar struct {
    char1 byte // a char
    int2 int // an integer
}

// a struct
type A struct {
    listInStruct []int // a list in a struct
    structInStruct StructWithAChar // a struct in a struct
}

// a sized struct
type SizedStruct struct {
    size int // the size
    stringInStruct string // the string
}

// emptyList: an empty list
// bufferString: here to check correct parsing of empty line above
// n: an integer, will be 0 in the sample input
// emptyInSample: an empty list (only in the sample)
// emptyString: an empty string
// main_: an other buffer string
// emptyCharList: an empty char list
// nonEmptyCharList: an char list, non empty
// structWithEmptyLine: a struct containing an empty line, then a struct
// aSizedStruct: a sized struct containing an empty line
// finish: a string to finish
func emptyLines(emptyList []int, bufferString string, n int, emptyInSample []int, emptyString string, main_ string, emptyCharList []byte, nonEmptyCharList []byte, structWithEmptyLine A, aSizedStruct SizedStruct, finish string) {
    /* TODO Wow, lots of empty lines! */
}

func main() {
    scanner := bufio.NewScanner(os.Stdin)
    scanner.Buffer(make([]byte, 0, 64 * 1024), 25769803764)
    emptyList := make([]int, 0)
    scanner.Scan()
    for i, iValue := range strings.SplitN(scanner.Text(), " ", 0) {
        emptyList[i], _ = strconv.Atoi(iValue)
    }
    var bufferString string
    scanner.Scan()
    bufferString = scanner.Text()
    var n int
    scanner.Scan()
    n, _ = strconv.Atoi(scanner.Text())
    emptyInSample := make([]int, n)
    scanner.Scan()
    for i, iValue := range strings.SplitN(scanner.Text(), " ", n) {
        emptyInSample[i], _ = strconv.Atoi(iValue)
    }
    var emptyString string
    scanner.Scan()
    emptyString = scanner.Text()
    var main_ string
    scanner.Scan()
    main_ = scanner.Text()
    var emptyCharList []byte
    scanner.Scan()
    emptyCharList = scanner.Bytes()
    var nonEmptyCharList []byte
    scanner.Scan()
    nonEmptyCharList = scanner.Bytes()
    var structWithEmptyLine A
    structWithEmptyLine.listInStruct = make([]int, n)
    scanner.Scan()
    for i, iValue := range strings.SplitN(scanner.Text(), " ", n) {
        structWithEmptyLine.listInStruct[i], _ = strconv.Atoi(iValue)
    }
    scanner.Scan()
    fmt.Sscanf(scanner.Text(), "%c %d", &structWithEmptyLine.structInStruct.char1, &structWithEmptyLine.structInStruct.int2)
    var aSizedStruct SizedStruct
    scanner.Scan()
    aSizedStruct.size, _ = strconv.Atoi(scanner.Text())
    scanner.Scan()
    aSizedStruct.stringInStruct = scanner.Text()
    var finish string
    scanner.Scan()
    finish = scanner.Text()
    emptyLines(emptyList, bufferString, n, emptyInSample, emptyString, main_, emptyCharList, nonEmptyCharList, structWithEmptyLine, aSizedStruct, finish);
}
